examples/checkpointing/checkpoint.py

```diff
@@ -13,22 +13,28 @@ class CheckpointOperator(Operator):
                     automatically include these cached arguments in the call to the
                     underlying devito.Operator.apply().
     """
-    t_arg_names = {'t_start': 'time_s', 't_end': 'time_e'}
+    t_arg_names = {'t_start': 'time_m', 't_end': 'time_M'}
 
     def __init__(self, op, **kwargs):
         self.op = op
-        self.time_order = kwargs.pop('time_order', 1)
         self.args = kwargs
+        t = {self.t_arg_names['t_end']: 0}
+        op_default_args = self.op.arguments(**t)
```

Comment:
Doesn't argument allocate all defaults?, that seems a bit overkill to get just one value.

Reply:
Agreed, so eventually @navjotk opted for this change (we discussed it on Slack in #development) but probably we can do better. Open to contributions

Reply:
So I need a reliable way of telling an `Operator`, "run your 1st timestep" or any nth timestep like that. Since time indices are now completely unpredictable (some operators run from 0 to `n-1`, some from 1 to `n` etc.), I was looking for a way to query this behaviour from the `Operator`. This is the way I was suggested for doing this. I do agree it is overkill and I raised this point on Slack at the time.

Reply:
Maybe we should split `arg_dims` and `arg_function` then to avoid memory overhead and get default loop indices/values?